persistence/pre_persistence/fisherman.go

```diff
@@ -0,0 +1,302 @@
+	if _, exists, _ := m.GetFisherman(address); exists {
+		return fmt.Errorf("already exists in world state")
+	}
+	cdc := Cdc()
+	db := m.Store()
+	key := append(FishermanPrefixKey, address...)
+	fish := Fisherman{
+		Address:         address,
+		PublicKey:       publicKey,
+		Paused:          paused,
+		Status:          int32(status),
+		Chains:          chains,
+		ServiceURL:      serviceURL,
+		StakedTokens:    stakedTokens,
+		PausedHeight:    uint64(pausedHeight),
+		UnstakingHeight: unstakingHeight,
+		Output:          output,
+	}
+	bz, err := cdc.Marshal(&fish)
+	if err != nil {
+		return err
+	}
+	return db.Put(key, bz)
+}
+
+func (m *PrePersistenceContext) UpdateFisherman(address []byte, serviceURL string, amountToAdd string, chains []string) error {
+	fish, exists, _ := m.GetFisherman(address)
+	if !exists {
+		return fmt.Errorf("does not exist in world state")
+	}
+	cdc := Cdc()
+	db := m.Store()
+	key := append(FishermanPrefixKey, address...)
+	// compute new values
+	stakedTokens, err := StringToBigInt(fish.StakedTokens)
+	if err != nil {
+		return err
+	}
+	stakedTokensToAddI, err := StringToBigInt(amountToAdd)
+	if err != nil {
+		return err
+	}
+	stakedTokens.Add(stakedTokens, stakedTokensToAddI)
+	// update values
+	fish.ServiceURL = serviceURL
+	fish.StakedTokens = BigIntToString(stakedTokens)
+	fish.Chains = chains
+	// marshal
+	bz, err := cdc.Marshal(fish)
+	if err != nil {
+		return err
+	}
+	return db.Put(key, bz)
+}
+
+func (m *PrePersistenceContext) DeleteFisherman(address []byte) error {
+	if exists, _ := m.GetFishermanExists(address); !exists {
+		return fmt.Errorf("does not exist in world state")
+	}
+	db := m.Store()
+	key := append(FishermanPrefixKey, address...)
+	return db.Put(key, DeletedPrefixKey)
+}
+
+func (m *PrePersistenceContext) GetFishermanReadyToUnstake(height int64, status int) (Fisherman []*types.UnstakingActor, err error) {
+	db := m.Store()
+	unstakingKey := append(UnstakingFishermanPrefixKey, []byte(fmt.Sprintf("%d", height))...)
+	if has := db.Contains(unstakingKey); !has {
+		return nil, nil
+	}
+	val, err := db.Get(unstakingKey)
+	if err != nil {
+		return nil, err
+	}
+	if val == nil {
+		return make([]*types.UnstakingActor, 0), nil
+	}
+	unstakingActors := types.UnstakingActors{}
+	if err := proto.Unmarshal(val, &unstakingActors); err != nil {
+		return nil, err
+	}
+	for _, sn := range unstakingActors.UnstakingActors {
+		Fisherman = append(Fisherman, sn)
+	}
+	return
+}
+
+func (m *PrePersistenceContext) GetFishermanStatus(address []byte) (status int, err error) {
+	fish, exists, err := m.GetFisherman(address)
+	if err != nil {
+		return ZeroInt, err
+	}
+	if !exists {
+		return ZeroInt, fmt.Errorf("does not exist in world state")
+	}
+	return int(fish.Status), nil
+}
+
+func (m *PrePersistenceContext) SetFishermanUnstakingHeightAndStatus(address []byte, unstakingHeight int64, status int) error {
```

Comment:
Figured it's worth pointing out that postgres does [support enums](https://www.postgresql.org/docs/9.1/datatype-enum.html), but It would require a lengthier discussion as to when/if/how the int <> enum conversion is done.

Reply:
Yup, this will take shape as the code evolves and adapts to datastores; at this point the observation and having it in our collective psyche is good enough.